### PR TITLE
8292948: JEditorPane ignores font-size styles in external linked css-file

### PR DESCRIPTION
**Pull request by stanio**
[JDK-8292948] reveals a regression by the fix for [JDK-8257664] (#1759) causing `font-size` declarations from external style sheets to be ignored.

As far as I've traced, the problem is with the `isDefined(CSS.Attribute.FONT_SIZE)` test:

https://github.com/openjdk/jdk/blob/70b5b3119b2ed032b021a080f34a4fa28d092fb5/src/java.desktop/share/classes/javax/swing/text/html/StyleSheet.java#L2825-L2830

It misses declarations from linked styles via a `resolver` ([`AttributeSet.ResolveAttribute`](https://github.com/openjdk/jdk/blob/b0e0b87891eb81c2b33c1cfa598701b7bd2e5bdf/src/java.desktop/share/classes/javax/swing/text/AttributeSet.java#L187-L191)) attribute that happens to hold the external style sheet rules.  The fix for this would be to move the implied `CSS.Attribute.FONT_SIZE` handling after:

https://github.com/openjdk/jdk/blob/70b5b3119b2ed032b021a080f34a4fa28d092fb5/src/java.desktop/share/classes/javax/swing/text/html/StyleSheet.java#L2833-L2835

and drop the `isDefined()` test.

\[I'll prepare a jtreg test further.\]

[JDK-8292948]: https://bugs.openjdk.org/browse/JDK-8292948 "JEditorPane ignores font-size styles in external linked css-file"
[JDK-8257664]: https://bugs.openjdk.org/browse/JDK-8257664 "HTMLEditorKit: Wrong CSS relative font sizes"

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292948](https://bugs.openjdk.org/browse/JDK-8292948): JEditorPane ignores font-size styles in external linked css-file

### Reviewers
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10054/head:pull/10054` \
`$ git checkout pull/10054`

Update a local copy of the PR: \
`$ git checkout pull/10054` \
`$ git pull https://git.openjdk.org/jdk pull/10054/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10054`

View PR using the GUI difftool: \
`$ git pr show -t 10054`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10054.diff">https://git.openjdk.org/jdk/pull/10054.diff</a>

</details>

### CODE REVIEW

File: test/jdk/javax/swing/text/html/StyleSheet/TestExternalCSSFontSize.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    private static final int[] expectedFontSizes = { 21, 14, 12 };
+
+    private JEditorPane editor;
+
+    private volatile Throwable failure;
+
+    TestExternalCSSFontSize() {}
+
+    CountDownLatch setUp() throws Exception {
+        File htmlFile = new File("TestExternalCSSFontSize.html");
```
Comment (stanio) [inline test/jdk/javax/swing/text/html/StyleSheet/TestExternalCSSFontSize.java:58]: Not sure if I should better resolve this from the classpath:

```suggestion
        URL htmlFile = getClass().getResource(getClass().getSimpleName() + ".html");
```

**Comment by stanio** [comment]
| Failure (current) | Success |
| - | - |
| failure | success |

**Review by prsadhuk — APPROVED** [review]

**Comment by stanio** [comment]
If there are no other reviewers, @prsadhuk, would you sponsor this change?

**Pull request closed without merging**